spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -3191,9 +3191,25 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
 
   private def partition2Proto(
       partition: FilePartition,
-      nativeScanBuilder: OperatorOuterClass.NativeScan.Builder): Unit = {
+      nativeScanBuilder: OperatorOuterClass.NativeScan.Builder,
+      scan: CometScanExec): Unit = {
     val partitionBuilder = OperatorOuterClass.SparkFilePartition.newBuilder()
+    val sparkContext = scan.session.sparkContext
+    var schema_saved: Boolean = false;
     partition.files.foreach(file => {
+      if (!schema_saved) {
+        // TODO: This code shouldn't be here, but for POC it's fine.
+        // Extract the schema and stash it.
+        val hadoopConf =
+          scan.relation.sparkSession.sessionState.newHadoopConfWithOptions(scan.relation.options)
+        val broadcastedHadoopConf =
+          sparkContext.broadcast(new SerializableConfiguration(hadoopConf))
+        val sharedConf = broadcastedHadoopConf.value.value
+        val footer = FooterReader.readFooter(sharedConf, file)
```

Comment:
You're right. This can never be in production code. For one, this is expensive.

Reply:
In theory it's just replacing the call that currently takes place during the file_reader instantiation, but yeah I'm still curious if it's already cached somewhere. I see references within Spark to a `footersCache` so I'm curious to look for that as well.

Reply:
I didn't know about a footersCache in Spark. Could you share a link maybe?

Reply:
It's actually part of parquet-hadoop: https://github.com/apache/parquet-java/blob/1e04ec74060a04cb029b6e61cf1098c1aba1446e/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetInputFormat.java#L525

Reply:
Ah, thanks. I don't think we ever travel this path.